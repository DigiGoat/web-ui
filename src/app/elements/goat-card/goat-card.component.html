<div class="card color-scheme-secondary" *ngIf="goat" test-id="goat">
  <div class="px-3 pt-3">
    <img *ngIf="image?.file" [ngSrc]="image!.file" [alt]="image?.alt" class="card-img-top rounded object-fit-contain" width="400"
      height="250" test-id="goat-image">
  </div>
  <div class="card-body color-scheme-secondary" [id]="identifier">
    <div class="container color-scheme-tertiary pb-3">
      <a *ngIf="identifier" class="stretched-link" [routerLink]="identifier" test-id="goat-link"
        attr.aria-label="Link to open {{identifier}}'s modal (This link cover's the whole element - it does not contain any text)"></a>
      <h5 class="card-title text-center mb-0" test-id="goat-name">{{name}}</h5>
      <h6 class="card-subtitle pb-1 fw-light fst-italic text-center" test-id="goat-age" *ngIf="born">{{born | age}}</h6>
      <h6 class="card-subtitle pb-1 text-center fw-light fst-italic font-monospace" *ngIf="linearAppraisal" test-id="goat-appraisal">
        {{linearAppraisal.generalAppearance}}{{linearAppraisal.dairyStrength}}{{linearAppraisal.bodyCapacity}}{{linearAppraisal.mammarySystem}}
        {{linearAppraisal.finalScore}}
      </h6>
      <h6 class="card-subtitle pb-1 text-center" *ngIf="nickname" test-id="goat-nickname">"{{nickname}}"</h6>
      <div class="card-text text-center text-truncate-3 pt-1" *ngIf="description" [innerHTML]="description" test-id="goat-description">
      </div>
    </div>
  </div>
</div>
<div class="card color-scheme-secondary" *ngIf="!goat" test-id="placeholder">
  <div class="px-3 pt-3 placeholder-glow">
    <img class="card-img-top rounded border placeholder" alt="Loading..." width="400" height="200" test-id="placeholder-image">
  </div>
  <div class="card-body color-scheme-secondary">
    <div class="container color-scheme-tertiary pb-3 text-center placeholder-wave">
      <h5 class="card-title text-center placeholder" test-id="placeholder-name">SOME ANIMALS NAME</h5>
      <br>
      <h6 class="card-subtitle pb-2 text-center placeholder" test-id="placeholder-nickname">"Nickname"</h6>
    </div>
  </div>
</div>