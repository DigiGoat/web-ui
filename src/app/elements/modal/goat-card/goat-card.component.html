<h5 class="card-title text-center" *ngIf="name" test-id="name">{{name}}</h5>
<h6 class="card-subtitle mb-2 text-center" *ngIf="nickname" test-id="nickname">"{{nickname}}"</h6>
<div class="card-header color-scheme-quaternary">
  <app-carousel [images]="images" test-id="carousel"></app-carousel>
  <div class="card-text text-center container color-scheme-tertiary rounded-bottom">
    <div class="text-capitalize">
      <div *ngIf="born">
        <span class="fw-bolder">Born</span>: <span class="fw-lighter" test-id="born">{{born | date:'longDate'}}</span>
        <br>
      </div>
      <div *ngIf="id">
        <span class="fw-bolder">ID</span>: <span class="fw-lighter" test-id="id">{{id}}</span>
        <br>
      </div>
      <div *ngIf="animalTattoos?.length">
        <span class="fw-bolder">Tattoo</span>: <span *ngFor="let animalTattoo of animalTattoos" test-id="tattoo">
          <span *ngIf="animalTattoo.tattooLocation?.name && animalTattoo.tattoo">
            {{animalTattoo.tattooLocation?.name}}:<span class="fw-lighter">{{animalTattoo.tattoo}}</span>
          </span>
        </span>
        <br>
      </div>
      <div *ngIf="colorAndMarking">
        <span class="fw-bolder">Markings</span>: <span class="fw-lighter" test-id="markings">{{colorAndMarking.toLowerCase()}}</span>
      </div>
    </div>
    <span test-id="description" *ngIf="description">{{description}}</span>
  </div>
</div>
