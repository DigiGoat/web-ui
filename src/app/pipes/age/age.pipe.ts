import { Pipe, PipeTransform } from '@angular/core';

@Pipe({
    name: 'age',
    standalone: false
})
export class AgePipe implements PipeTransform {

  transform(value: string): string {
    try {
      if (!Date.parse(value)) {
        throw new Error('Failed to parse date');
      }
      /** The targeted date in days */
      const date = Date.parse(value) / (1000 * 3600 * 24);
      /** The current date in days */
      const now = Date.now() / (1000 * 3600 * 24);
      const difference = now - date;
      if (difference < 0) {
        throw new Error('Provided date is in the future');
      }

      /** Check if the date is within the past week */
      if (difference < 7) {
        const days = Math.trunc(difference);
        value = `${days} day${days > 1 ? 's' : ''} old`;
        /** Check if the date is within the past 5 weeks */
      } else if (difference < 7 * 5) {
        const weeks = Math.trunc(difference / 7);
        value = `${weeks} week old`;
        /** Check if the date is within the past year */
      } else if (difference < 365) {
        const months = Math.trunc(difference / (365 / 12));
        value = `${months} month old`;
        /** Everything else is more than 1 year */
      } else {
        const years = Math.trunc(difference / 365);
        value = `${years} year old`;
      }
    } catch (e) {
      console.warn('Failed to parse age from date:', value, 'With error:', e);
    }
    return value;
  }

}
