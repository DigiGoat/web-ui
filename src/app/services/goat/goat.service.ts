import { Observable } from 'rxjs';
import { retry } from 'rxjs/operators';

import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';


@Injectable({
  providedIn: 'root'
})
export class GoatService {
  constructor(private http: HttpClient) { }
  private _does: Goat[] = [];
  public does = new Observable<Goat[]>(observer => {
    if (this._does.length) {
      console.debug('Loaded Does From Cache', this._does);
      observer.next(this._does);
    } else {
      this.http.get<Goat[]>('/assets/resources/does.json')
        .pipe(
          retry(3), // retry a failed request up to 3 times
        )
        .subscribe({
          next: data => {
            this._does = data;
            console.debug('Loaded Does From Server', data);
            observer.next(data);
          },
          error: err => {
            if (err.status === 0) {
              // A client-side or network error occurred. Handle it accordingly.
              console.warn('An error occurred:', err.error);
            } else {
              // The backend returned an unsuccessful response code.
              // The response body may contain clues as to what went wrong.
              console.warn(
                `Backend returned code ${err.status}, body was: `, err.error);
            }
            // Return an observable with a user-facing error message.
            observer.error(err);
          }
        });
    }
  });
  private _bucks: Goat[] = [];
  public bucks = new Observable<Goat[]>(observer => {
    if (this._bucks.length) {
      console.debug('Loaded Bucks From Cache', this._bucks);
      observer.next(this._bucks);
    } else {
      this.http.get<Goat[]>('/assets/resources/bucks.json')
        .pipe(
          retry(3), // retry a failed request up to 3 times
        )
        .subscribe({
          next: data => {
            this._bucks = data;
            console.debug('Loaded Bucks From Server', data);
            observer.next(data);
          },
          error: err => {
            if (err.status === 0) {
              // A client-side or network error occurred. Handle it accordingly.
              console.warn('An error occurred:', err.error);
            } else {
              // The backend returned an unsuccessful response code.
              // The response body may contain clues as to what went wrong.
              console.warn(
                `Backend returned code ${err.status}, body was: `, err.error);
            }
            // Return an observable with a user-facing error message.
            observer.error(err);
          }
        });
    }
  });
}
//export type Goat = (OwnedGoats['result']['items'][number] & { nickname: string; description: string; awards: Awards['result']['items']; colorAndMarking: string; /*obtained?: string;*/ });
export type Goat = {
  nickname?: string;
  name: string;
  description?: string;
  normalizeId: string;
  dateOfBirth: string;
  colorAndMarking?: string;
  animalTattoo?: [{ tattoo?: string; tattooLocation?: { name?: string; }; }, { tattoo?: string; tattooLocation?: { name?: string; }; }];
};
export const Goat = {
  nickname: 'Your Goats Farm Name',
  name: 'YOUR GOATS ADGA NAME',
  description: 'A quick description about your goat',
  normalizeId: 'PD12345',
};
