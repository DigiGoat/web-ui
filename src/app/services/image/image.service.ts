import { Injectable } from '@angular/core';

import imageMap from '../../../assets/images/map.json';


@Injectable({
  providedIn: 'root'
})
export class ImageService {
  private imageMap: ImageMap = imageMap;

  constructor() { }
  getImage(searchQueries: (string | undefined)[]) {
    const key = Object.keys(this.imageMap).find(directory => searchQueries.includes(directory));
    if (key && this.imageMap[key].length) {
      const image = this.imageMap[key][0];
      return { ...image, file: `/assets/images/${key}/${image.file}` };
    }
    //Intentionally give an invalid link
    return { ...this.NotFound, file: '/' };
  }
  getImages(searchQueries: (string | undefined)[]) {
    const key = Object.keys(this.imageMap).find(directory => searchQueries.includes(directory));
    if (key && this.imageMap[key].length) {
      const images = this.imageMap[key].map(image => { return { ...image, file: `/assets/images/${key}/${image.file}` }; });
      return images;
    }
    //Intentionally give an invalid link
    return [{ ...this.NotFound, file: '/' }];
  }

  public readonly NotFound: ImageEntry = { file: '/assets/images/ImageNotFound.png', alt: 'The Requested Image Does Not Exist' };
}
type ImageMap = {
  [directory: string]: ImageEntry[];
};
export type ImageEntry = {
  file: string,
  alt?: string;
};
