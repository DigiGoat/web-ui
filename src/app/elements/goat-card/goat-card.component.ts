import { Component, Input, type OnChanges } from '@angular/core';
import { Meta } from '@angular/platform-browser';
import { ActivatedRoute } from '@angular/router';
import { type Goat, GoatService } from '../../services/goat/goat.service';
import { ConfigService } from '../../services/config/config.service';
import { ImageEntry, ImageService } from '../../services/image/image.service';
@Component({
    selector: 'app-goat-card',
    templateUrl: './goat-card.component.html',
    styleUrls: ['./goat-card.component.scss'],
    standalone: false
})
export class GoatCardComponent implements OnChanges {
  constructor(private imageService: ImageService, private meta: Meta, private configService: ConfigService, private route: ActivatedRoute, private goatService: GoatService) { }

  @Input() goat?: Partial<Goat>;
  name?: string;
  nickname?: string;
  description?: string;
  id?: string;
  born?: string;
  image?: ImageEntry;
  identifier?: string;
  linearAppraisal?: Required<Goat>['linearAppraisals'][number];
  ngOnChanges() {
    this.name = this.goat?.name;
    this.nickname = this.goat?.nickname;
    this.description = this.goat?.description;
    this.id = this.goat?.normalizeId;
    this.born = this.goat?.dateOfBirth;
    this.image = this.imageService.getImage([this.id, this.name, this.nickname]);
    this.identifier = this.nickname ?? this.name?.replace(/ /g, '-') ?? this.id;
    if (!this.route.snapshot.params['goat']) {
      this.setOGImages();
    }
    this.linearAppraisal = this.goatService.getAppraisal(this.goat?.linearAppraisals);
  }
  setOGImages() {
    if (this.image && this.image.file !== '/') {
      if (!this.meta.getTags('property="og:image"').find(tag => tag.content.endsWith(this.image!.file))) {
        this.meta.addTag({ property: 'og:image', content: this.configService.link ? new URL(this.image.file, this.configService.link).toString() : this.image.file });
        //this.meta.addTags([{ name: 'og:image:width', content: '400' }, { name: 'og:image:height', content: '200' }], true);
        if (this.image.alt) {
          this.meta.addTag({ property: 'og:image:alt', content: this.image.alt });
        }
      }
    }
  }
}
