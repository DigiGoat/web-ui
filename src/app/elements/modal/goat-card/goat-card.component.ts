import { Component, Input, type OnInit } from '@angular/core';
import { Meta } from '@angular/platform-browser';
import type { Observable } from 'rxjs';
import type { Page } from '../../../app-routing.module';
import type { Goat } from '../../../services/goat/goat.service';
import { ImageService, type ImageEntry } from '../../../services/image/image.service';

@Component({
  selector: 'app-modal-goat-card',
  templateUrl: './goat-card.component.html',
  styleUrl: './goat-card.component.scss'
})
export class GoatCardComponent implements OnInit, Page {
  @Input({ required: true }) goat?: Partial<Goat>;

  constructor(public imageService: ImageService, private meta: Meta) { }
  setDescription(): void | Observable<void> {
    if (this.description) {
      this.meta.addTags([{ name: 'og:description', content: this.description }, { name: 'description', content: this.description }]);
    }
  }

  name?: string;
  nickname?: string;
  description?: string;
  id?: string;
  born?: string | number;
  animalTattoos?: Goat['animalTattoo'];
  colorAndMarking?: string;
  images!: ImageEntry[];
  ngOnInit(): void {
    this.name = this.goat?.name;
    this.nickname = this.goat?.nickname;
    this.description = this.goat?.description;
    this.id = this.goat?.normalizeId;
    this.born = this.goat?.dateOfBirth;
    this.animalTattoos = this.goat?.animalTattoo;
    this.colorAndMarking = this.goat?.colorAndMarking;
    this.images = this.imageService.getImages([this.id, this.name, this.nickname]);
    this.setDescription();
  }
}
