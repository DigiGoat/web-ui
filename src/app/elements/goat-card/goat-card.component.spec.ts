import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ActivatedRoute } from '@angular/router';
import { AgePipe } from '../../pipes/age/age.pipe';
import { GoatService } from '../../services/goat/goat.service';
import { ImageService } from '../../services/image/image.service';
import { GoatCardComponent } from './goat-card.component';

jest.mock('@angular/router');
jest.mock('../../services/goat/goat.service');

describe('GoatCardComponent', () => {
  let component: GoatCardComponent;
  let fixture: ComponentFixture<GoatCardComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [GoatCardComponent, AgePipe],
      providers: [ImageService, ActivatedRoute, GoatService],
    })
      .compileComponents();
    fixture = TestBed.createComponent(GoatCardComponent);
    component = fixture.componentInstance;
    TestBed.inject(ImageService);
    component['imageService']['imageMap'] = {
      TEST_NICKNAME: [
        { file: 'TEST_IMAGE_FILE', alt: 'TEST_IMAGE_ALT' }
      ]
    };
    component['route'] = { snapshot: { params: {} } } as any;
    component['goatService']['getAppraisal'] = jest.fn().mockReturnValue({ finalScore: 80 }) as any;
  });

  it('should create', () => {
    fixture.detectChanges();
    expect(component).toBeTruthy();
  });
  describe('With a Goat (v1.0.0)', () => {
    let html: HTMLElement;
    beforeEach(() => {
      component['goat'] = {
        nickname: 'TEST_NICKNAME',
        name: 'YOUR GOATS ADGA NAME',
        description: 'A quick description about your goat',
        dateOfBirth: new Date(Date.now() - (1000 * 3600 * 24)).toString(),
        normalizeId: 'PD12345',
      };
      component.ngOnChanges();
      fixture.detectChanges();
      html = fixture.nativeElement;
    });
    it('should exist', () => {
      const element = html.querySelector('[test-id="goat"]') as HTMLDivElement;
      expect(element).toBeTruthy();
    });
    it('should NOT display the placeholder div', () => {
      const element = html.querySelector('[test-id="placeholder"]') as HTMLDivElement;
      expect(element).toBeFalsy();
    });
    it('should have an image', () => {
      const element = html.querySelector('[test-id="goat-image"]') as HTMLImageElement & { ngSrc: string; };
      expect(element).toBeTruthy();
      expect(element.ngSrc).toBe('./assets/images/TEST_NICKNAME/TEST_IMAGE_FILE');
      expect(element.width).toBe(400);
      expect(element.height).toBe(250);
    });
    it('should have a link', () => {
      const element = html.querySelector('[test-id="goat-link"]') as HTMLAnchorElement & { routerLink: string; };
      expect(element).toBeTruthy();
      expect(element.routerLink).toBe('TEST_NICKNAME');
    });
    it('should have a name', () => {
      const element = html.querySelector('[test-id="goat-name"]') as HTMLHeadingElement;
      expect(element).toBeTruthy();
      expect(element.innerHTML).toBe('YOUR GOATS ADGA NAME');
    });
    it('should have an age', () => {
      const element = html.querySelector('[test-id="goat-age"]') as HTMLHeadingElement;
      expect(element).toBeTruthy();
      expect(element.innerHTML).toBe('1 day old');
    });
    it('should have a nickname', () => {
      const element = html.querySelector('[test-id="goat-nickname"]') as HTMLHeadingElement;
      expect(element).toBeTruthy();
      expect(element.innerHTML).toBe('"TEST_NICKNAME"');
    });
    it('should have a description', () => {
      const element = html.querySelector('[test-id="goat-description"]') as HTMLDivElement;
      expect(element).toBeTruthy();
      expect(element.innerHTML).toBe('A quick description about your goat');
    });
    it('should match the snapshot', () => {
      expect(fixture).toMatchSnapshot();
    });
  });
  describe('With an empty Goat', () => {
    let html: HTMLElement;
    beforeEach(() => {
      component['goat'] = {};
      component.ngOnChanges();
      fixture.detectChanges();
      html = fixture.nativeElement;
    });
    it('should exist', () => {
      const element = html.querySelector('[test-id="goat"]') as HTMLDivElement;
      expect(element).toBeTruthy();
    });
    it('should NOT display the placeholder div', () => {
      const element = html.querySelector('[test-id="placeholder"]') as HTMLDivElement;
      expect(element).toBeFalsy();
    });
    it('should have an image', () => {
      const element = html.querySelector('[test-id="goat-image"]') as HTMLImageElement & { ngSrc: string; };
      expect(element).toBeTruthy();
      expect(element.ngSrc).toBe('/');
      expect(element.width).toBe(400);
      expect(element.height).toBe(250);
    });
    it('should have a link', () => {
      const element = html.querySelector('[test-id="goat-link"]') as HTMLAnchorElement & { routerLink: string; };
      expect(element).toBeFalsy();
    });
    it('should have a name', () => {
      const element = html.querySelector('[test-id="goat-name"]') as HTMLHeadingElement;
      expect(element).toBeTruthy();
      expect(element.innerHTML).toBe('');
    });
    it('should have an age', () => {
      const element = html.querySelector('[test-id="goat-age"]') as HTMLHeadingElement;
      expect(element).toBeFalsy();
    });
    it('should have a nickname', () => {
      const element = html.querySelector('[test-id="goat-nickname"]') as HTMLHeadingElement;
      expect(element).toBeFalsy();
    });
    it('should have a description', () => {
      const element = html.querySelector('[test-id="goat-description"]') as HTMLDivElement;
      expect(element).toBeFalsy();
    });
    it('should match the snapshot', () => {
      expect(fixture).toMatchSnapshot();
    });
  });
  describe('Without a Goat', () => {
    let html: HTMLElement;
    beforeEach(() => {
      fixture.detectChanges();
      html = fixture.nativeElement;
    });
    it('should exist', () => {
      const element = html.querySelector('[test-id="placeholder"]') as HTMLDivElement;
      expect(element).toBeTruthy();
    });
    it('should NOT display the goat div', () => {
      const element = html.querySelector('[test-id="goat"]') as HTMLDivElement;
      expect(element).toBeFalsy();
    });
    it('should have a placeholder image', () => {
      const element = html.querySelector('[test-id="placeholder-image"]') as HTMLImageElement;
      expect(element).toBeTruthy();
      expect(element.parentElement!.classList).toContain('placeholder-glow');
      expect(element.width).toBe(400);
      expect(element.height).toBe(200);
    });
    it('should have a placeholder name', () => {
      const element = html.querySelector('[test-id="placeholder-name"]') as HTMLHeadingElement;
      expect(element).toBeTruthy();
      expect(element.classList).toContain('placeholder');
      expect(element.parentElement!.classList).toContain('placeholder-wave');
      expect(element.textContent).toBe('SOME ANIMALS NAME');
    });
    it('should have a placeholder nickname', () => {
      const element = html.querySelector('[test-id="placeholder-nickname"]') as HTMLHeadingElement;
      expect(element).toBeTruthy();
      expect(element.classList).toContain('placeholder');
      expect(element.parentElement!.classList).toContain('placeholder-wave');
      expect(element.textContent).toBe('"Nickname"');
    });
    it('should match the snapshot', () => {
      expect(fixture).toMatchSnapshot();
    });
  });
});
