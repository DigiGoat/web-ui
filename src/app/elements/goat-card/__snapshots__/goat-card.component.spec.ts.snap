// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoatCardComponent With a Goat (v1.0.0) should match the snapshot 1`] = `
<app-goat-card
  born={[Function String]}
  configService={[Function ConfigService]}
  description={[Function String]}
  goat={[Function Object]}
  goatService={[Function GoatService]}
  id={[Function String]}
  identifier={[Function String]}
  image={[Function Object]}
  imageService={[Function ImageService]}
  linearAppraisal={[Function Object]}
  meta={[Function _Meta]}
  name={[Function String]}
  nickname={[Function String]}
  route={[Function Object]}
>
  <div
    class="card color-scheme-secondary"
    test-id="goat"
  >
    <div
      class="px-3 pt-3"
    >
      <img
        alt="TEST_IMAGE_ALT"
        class="card-img-top rounded object-fit-contain"
        height="250"
        test-id="goat-image"
        width="400"
      />
    </div>
    <div
      class="card-body color-scheme-secondary"
      id="TEST_NICKNAME"
    >
      <div
        class="container color-scheme-tertiary pb-3"
      >
        <a
          aria-label="Link to open TEST_NICKNAME's modal (This link cover's the whole element - it does not contain any text)"
          class="stretched-link"
          test-id="goat-link"
        />
        <h5
          class="card-title text-center mb-0"
          test-id="goat-name"
        >
          YOUR GOATS ADGA NAME
        </h5>
        <h6
          class="card-subtitle pb-1 fw-light fst-italic text-center"
          test-id="goat-age"
        >
          1 day old
        </h6>
        <h6
          class="card-subtitle pb-1 text-center fw-light fst-italic font-monospace"
          test-id="goat-appraisal"
        >
            80 
        </h6>
        <h6
          class="card-subtitle pb-1 text-center"
          test-id="goat-nickname"
        >
          "TEST_NICKNAME"
        </h6>
        <div
          class="card-text text-center text-truncate-3 pt-1"
          test-id="goat-description"
        >
          A quick description about your goat
        </div>
      </div>
    </div>
  </div>
</app-goat-card>
`;

exports[`GoatCardComponent With an empty Goat should match the snapshot 1`] = `
<app-goat-card
  born="undefined"
  configService={[Function ConfigService]}
  description="undefined"
  goat={[Function Object]}
  goatService={[Function GoatService]}
  id="undefined"
  identifier="undefined"
  image={[Function Object]}
  imageService={[Function ImageService]}
  linearAppraisal={[Function Object]}
  meta={[Function _Meta]}
  name="undefined"
  nickname="undefined"
  route={[Function Object]}
>
  <div
    class="card color-scheme-secondary"
    test-id="goat"
  >
    <div
      class="px-3 pt-3"
    >
      <img
        alt="The Requested Image Does Not Exist"
        class="card-img-top rounded object-fit-contain"
        height="250"
        test-id="goat-image"
        width="400"
      />
    </div>
    <div
      class="card-body color-scheme-secondary"
      id="undefined"
    >
      <div
        class="container color-scheme-tertiary pb-3"
      >
        <h5
          class="card-title text-center mb-0"
          test-id="goat-name"
        >
        </h5>
        <h6
          class="card-subtitle pb-1 text-center fw-light fst-italic font-monospace"
          test-id="goat-appraisal"
        >
            80 
        </h6>
      </div>
    </div>
  </div>
</app-goat-card>
`;

exports[`GoatCardComponent Without a Goat should match the snapshot 1`] = `
<app-goat-card
  configService={[Function ConfigService]}
  goatService={[Function GoatService]}
  imageService={[Function ImageService]}
  meta={[Function _Meta]}
  route={[Function Object]}
>
  <div
    class="card color-scheme-secondary"
    test-id="placeholder"
  >
    <div
      class="px-3 pt-3 placeholder-glow"
    >
      <img
        alt="Loading..."
        class="card-img-top rounded border placeholder"
        height="200"
        test-id="placeholder-image"
        width="400"
      />
    </div>
    <div
      class="card-body color-scheme-secondary"
    >
      <div
        class="container color-scheme-tertiary pb-3 text-center placeholder-wave"
      >
        <h5
          class="card-title text-center placeholder"
          test-id="placeholder-name"
        >
          SOME ANIMALS NAME
        </h5>
        <br />
        <h6
          class="card-subtitle pb-2 text-center placeholder"
          test-id="placeholder-nickname"
        >
          "Nickname"
        </h6>
      </div>
    </div>
  </div>
</app-goat-card>
`;
